sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager_base.py

```diff
@@ -25,24 +26,104 @@
     ETAG_KEY,
     FEATURE_FLAG_REFERENCE_KEY,
     FEATURE_FLAG_ID_KEY,
+    ALLOCATION_ID_KEY,
 )
 
 FALLBACK_CLIENT_REFRESH_EXPIRED_INTERVAL = 3600  # 1 hour in seconds
 MINIMAL_CLIENT_REFRESH_INTERVAL = 30  # 30 seconds
 
+JSON = Mapping[str, Any]
+
 
 @dataclass
 class _ConfigurationClientWrapperBase:
     endpoint: str
 
+    @staticmethod
+    def _generate_allocation_id(feature_flag_value: Dict[str, JSON]) -> Optional[str]:
+        """
+        Generates an allocation ID for the specified feature.
+        seed=123abc\ndefault_when_enabled=Control\npercentiles=0,Control,20;20,Test,100\nvariants=Control,standard;Test,special # pylint:disable=line-too-long
+
+        :param  Dict[str, JSON] feature_flag_value: The feature to generate an allocation ID for.
+        :rtype: str
+        :return: The allocation ID.
+        """
+
+        allocation_id = ""
+        allocated_variants = []
+
+        allocation: Optional[JSON] = feature_flag_value.get("allocation")
+
+        if allocation:
+            # Seed
+            allocation_id = f"seed={allocation.get('seed', '')}"
+
+            # DefaultWhenEnabled
+            if "default_when_enabled" in allocation:
+                allocated_variants.append(allocation.get("default_when_enabled"))
+
+            allocation_id += f"\ndefault_when_enabled={allocation.get('default_when_enabled', '')}"
```

Comment:
why isn't default_when_enabled base64 encoded here (just like what you did for other variants in percentile allocation)

Reply:
I can't think of a value that would cause a collision here- but for consistency I wouldn't be opposed to adding it.

Reply:
If we update this than this will need to be updated too: https://github.com/Azure/AppConfiguration-DotnetProvider/blob/c2e18d7b2c2c458e6951fc88f17e1ec0ebc13884/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs#L355